hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DisableTableProcedure.java

```diff
@@ -219,7 +233,16 @@ protected void deserializeStateData(ProcedureStateSerializer serializer) throws
   // SCP.
   @Override
   protected boolean holdLock(MasterProcedureEnv env) {
-    return true;
+    // We will test holdLock when releaseLock, if we are still at the first state, it means we fail
+    // the SCP check and want to retry later, so we should release the lock
+    return getCurrentState() != DisableTableState.DISABLE_TABLE_PREPARE;
```

Comment:
we would hold the lock at DISABLE_TABLE_PRE_OPERATION still right? what if the SCP gets created (and creates ASSIGN procs also) at that time?

Reply:
More or less was pointing the same only, in current approach we are waiting for the SCP to finish but the table state is still not DISABLING yet, 
so I was saying in DISABLE_TABLE_PRE_OPERATION (which is next state) since we are not waiting here, SCP could be again created at this time and we can again see this issue.

```
Then I think first we need to introduce another mechanism to only allow one Enable/Modify/DisableTableProcedure at the same time, then we can be back here to change the i 
for DisableTableProcedure.
```
But then again, instead of releasing locks or handling locks for these Tablelevel procedures separeately, if we share the lock with SCP, that also works well know?

Like if we see, everytime only SCP can interrupt any ongoing procedure,
Which ever procedure is waiting for an existing procedure to complete, it is kind of dependent on it and makes sense to share the lock of resources it holds, (since its waiting on that procedure anyway).
And we push the current retry strategy at procedure framework level @Apache9